crates/ty_python_semantic/src/types/signatures.rs

```diff
@@ -197,6 +201,69 @@ impl<'db> CallableSignature<'db> {
         tcx: TypeContext<'db>,
         visitor: &ApplyTypeMappingVisitor<'db>,
     ) -> Self {
+        if let TypeMapping::Specialization(specialization) = type_mapping
```

Comment:
We'll need this logic for `TypeMapping::PartialSpecialization` too. It might be worth pulling much of the body out into a helper method.

Reply:
Ah yes, I've extracted the common part as a nested function in this function body.